src/dmd/builtin.d

```diff
@@ -103,6 +103,116 @@ extern (C++) Expression eval_ldexp(Loc loc, FuncDeclaration fd, Expressions* arg
     return new RealExp(loc, CTFloat.ldexp(arg0.toReal(), cast(int) arg1.toInteger()), arg0.type);
 }
 
+extern (C++) Expression eval_log(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.log(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_log2(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.log2(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_log10(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.log10(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_expm1(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.expm1(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_exp2(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.exp2(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_round(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.round(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_floor(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.floor(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_ceil(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.ceil(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_trunc(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.trunc(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_copysign(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    Expression arg1 = (*arguments)[1];
+    assert(arg1.op == TOK.float64);
+    return new RealExp(loc, CTFloat.copysign(arg0.toReal(), cast(int) arg1.toInteger()), arg0.type);
```

Comment:
LDC fails to comple this line (no implicit conversion to real_t). Why do you cast to int here? copysign expects a real parameter.

Reply:
@rainers - Doesn't ldc implement their own builtins module independent of dmd?

Reply:
I think so, but as noted, this is a usage of struct longdouble introduced by compilation with LDC. AFAICT passing `int` here is wrong anyway, should just be `arg1.toReal()`.